Core/include/Acts/TrackFitting/detail/GsfActor.hpp

```diff
@@ -317,13 +317,20 @@ struct GsfActor {
       applyMultipleScattering(state, stepper, navigator,
                               MaterialUpdateStage::PostUpdate);
     }
+  }
 
-    // Break the navigation if we found all measurements
+  template <typename propagator_state_t, typename stepper_t,
+            typename navigator_t>
+  bool checkAbort(propagator_state_t& /*state*/, const stepper_t& /*stepper*/,
```

Comment:
This contracts the main actor with the separate aborter, which is possible after you merged the two lists, right?

Reply:
Yes exactly there is not reason to separate them anymore. I did this on the side after realizing the abort condition was communicated through the navigator state which is really not ideal. This could be factored out of this PR in principle

Reply:
Your call.